plugins/modules/keyring_info.py

```diff
@@ -81,10 +81,9 @@
 
 
 def _alternate_retrieval_method(module):
-    get_argument = 'echo "%s" | gnome-keyring-daemon --unlock\nkeyring get %s %s\n' % (
-        quote(module.params["keyring_password"]),
-        quote(module.params["service"]),
-        quote(module.params["username"]),
+    get_argument = (
+        f'echo "{quote(module.params["keyring_password"])}" | gnome-keyring-daemon --unlock\n'
```

Comment:
I'm wondering whether the double quotes around the `quote()` result are a bug, or a (failed) attempt at double quoting. In any case, this isn't a bug caused by this PR, but this should be investigated.

Reply:
AFAICT, it is much less sophisticated than that. It is literally replacing the `%s` part of `"%s"` with `{`_the expression in the list_`}`, hence it becomes `"{quote(module.params["keyring_password"])}"`.

The handling of quotes by `flynt` is less than stellar - it still beats scanning and replacing with f-str expressions manually, though.

Reply:
This isn't related to flynt, the old code also was doing that. The output is a double quote followed by the result of `quote(...)` followed by another double quote, which depending on the value of `module.params["keyring_password"]` screws up the value. For example, if the keyring password happens to be `foo 'bar'`, then the resulting command is  `echo "'foo '"'"'bar'"'"''"`, which results in the output of `'foo '"bar"''`. I don't know what kind of input `gnome-keyring-daemon` expects, but this likely makes it fail, or passes the wrong password.